src/pudl/transform/vcerare.py

```diff
@@ -0,0 +1,330 @@
+    """Combine capacity factor tables."""
+    logger.info("Merging all the capacity factor tables into one")
+    merge_keys = ["report_year", "datetime_utc", "hour_of_year", "county_state_names"]
+    for name, df in cap_fac_dict.items():
+        cap_fac_dict[name] = df.set_index(merge_keys)
+    mega_df = pd.concat(
+        [
+            cap_fac_dict["solar_pv"],
+            cap_fac_dict["offshore_wind"],
+            cap_fac_dict["onshore_wind"],
+        ],
+        axis="columns",
+    ).reset_index()
+
+    return mega_df
+
+
+def _combine_cap_fac_with_fips_df(
+    cap_fac_df: pd.DataFrame, fips_df: pd.DataFrame
+) -> pd.DataFrame:
+    """Combine the combined capacity factor df with the fips df."""
+    logger.info(
+        "Merging the combined capacity factor table with the Lat-Long-FIPS table"
+    )
+    combined_df = pd.merge(cap_fac_df, fips_df, on="county_state_names", how="left")
+    return combined_df
+
+
+def _combine_city_county_records(df: pd.DataFrame) -> pd.DataFrame:
+    """Average cap fac values for city and county values with the same FIPS ID.
+
+    There are several duplicate FIPS code values in the data. Most of them
+    are lakes within a county, but some of them are cities. Only one of the
+    cities, clifton forge city, has values, so we average those values
+    with that of the county. The other city, XXX, has no values so we
+    remove it.
+
+    """
+    bad_county_mask = df["county_state_names"].isin(
+        ["alleghany_virginia", "clifton_forge_city_virginia"]
+    )
+    # Take the subset of the data that's relevant to the changes and sort the values
+    # by county_state so that alleghany is first and those are the values that are
+    # preserved by the groupby "first" below.
+    subset_df = df[bad_county_mask].sort_values(by=["county_state_names"])
+    cap_fac_cols = [x for x in subset_df.columns if "capacity_factor" in x]
+    other_cols = [
+        x for x in subset_df.columns if x not in cap_fac_cols + ["datetime_utc"]
+    ]
+    grouped_df = (
+        subset_df.groupby(["datetime_utc"], sort=False)
+        .agg(
+            {
+                **{col: "mean" for col in cap_fac_cols},
+                **{col: "first" for col in other_cols},
+            }
+        )
+        .reset_index()
+    )
+    out_df = pd.concat([df[~bad_county_mask], grouped_df])
+    # Now lets remove the records for the other city with no values
+    bad_county_mask_2 = out_df["county_state_names"] == "bedford_city_virginia"
+    # Make sure there aren't any non-zero values
+    if not (out_df[bad_county_mask_2][cap_fac_cols] == 0).all().all():
+        raise AssertionError("Found non-zero values for bedford_city_virginia record.")
+    return out_df[~bad_county_mask_2]
+
+
+@asset(
+    io_manager_key="parquet_io_manager",
+    compute_kind="pandas",
+    op_tags={"memory-use": "high"},
+)
+def out_vcerare__hourly_available_capacity_factor(
+    raw_vcegen__lat_lon_fips: pd.DataFrame,
```

Comment:
```suggestion
    raw_vcerare__lat_lon_fips: pd.DataFrame,
```
This should get updated, right?